### PR TITLE
Final GitHub Action Pull Request

### PR DESCRIPTION
**Pull request by ConnorDownes**
This is to check whether the unit tests now pass when the workflow action runs.

### CODE REVIEW

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Node.js CI
+
+on: [push, pull_request]
```
Comment (SGibbsEG) [inline .github/workflows/nodejs.yml]: make this PR only. otherwise we will get spammed.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        node-version: [8.x, 10.x, 12.x]
```
Comment (SGibbsEG) [inline .github/workflows/nodejs.yml]: only care about latest versions 12.x

**Comment by ConnorDownes** [comment]
Updated based off feedback. Windows latest is erroring.
This appears to be because we are using `npm ci`. Switching to `npm i` will apparently sort this out.

"npm ci ignores the os field in package.json. I've raised a bug in the npm repo. fsevents@1.2.9 still works, because it downloads the executables. Where 1.2.11 does not, because it tries to compile them. When npm ci is run, the package is installed regardless of the os field, and node-gyp is always executed. This causes very old (npm@3) errors to reappear on Windows CI environments."
https://github.com/fsevents/fsevents/issues/301

**Comment by SGibbsEG** [comment]
> Updated based off feedback. Windows latest is erroring.
> This appears to be because we are using `npm ci`. Switching to `npm i` will apparently sort this out.
> 
> "npm ci ignores the os field in package.json. I've raised a bug in the npm repo. fsevents@1.2.9 still works, because it downloads the executables. Where 1.2.11 does not, because it tries to compile them. When npm ci is run, the package is installed regardless of the os field, and node-gyp is always executed. This causes very old (npm@3) errors to reappear on Windows CI environments."
> [fsevents/fsevents#301](https://github.com/fsevents/fsevents/issues/301)

In which case just make it so we are testing ubuntu. I am thinking of publishing to linux anyways as the hosting is way cheaper and we can use docker containers that way.

sack off windows and macos

**Comment by StephenGibbsDev** [comment]
> Updated based off feedback. Windows latest is erroring.
> This appears to be because we are using `npm ci`. Switching to `npm i` will apparently sort this out.
> 
> "npm ci ignores the os field in package.json. I've raised a bug in the npm repo. fsevents@1.2.9 still works, because it downloads the executables. Where 1.2.11 does not, because it tries to compile them. When npm ci is run, the package is installed regardless of the os field, and node-gyp is always executed. This causes very old (npm@3) errors to reappear on Windows CI environments."
> [fsevents/fsevents#301](https://github.com/fsevents/fsevents/issues/301)

sack off windows and macOS then. i am planning on only releasing on linux anyways. we can always add it back in once the bug is sorted.

**Comment by StephenGibbsDev** [comment]
@ConnorDownes 
So for some reason jest doesnt work on windows-latest therefore i have changed the workflow file to target windows-2016 instead

**Pull request merged**